flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PartitionRequestNotifier.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition;
+
+import org.apache.flink.runtime.io.network.NetworkSequenceViewReader;
+import org.apache.flink.runtime.io.network.partition.consumer.InputChannelID;
+
+import java.io.IOException;
+
+/**
+ * When the netty server receives a downstream task's partition request event and finds its upstream task doesn't register its partition yet,
+ * the netty server will construct a {@link PartitionRequestNotifier} and notify the request when the task deploys itself and
+ * registers its partition to {@link ResultPartitionManager}.
+ */
+public interface PartitionRequestNotifier {
```

Comment:
Shouldn't this be called `PartitionCreationListener`? This is NOT notifying about a partition request. But partition request is creating/using this "listener", when partition hasn't been yet created, to listener for partition creation notification.